src/Microsoft.PowerApps.TestEngine/TestInfra/PlaywrightTestInfraFunctions.cs

```diff
@@ -307,21 +307,78 @@ public async Task HandleUserEmailScreen(string selector, string value)
         [ExcludeFromCodeCoverage]
         public async Task HandleUserPasswordScreen(string selector, string value, string desiredUrl)
         {
+            var logger = _singleTestInstanceState.GetLogger();
+
+            // Setting options fot the RunAndWaitForNavigationAsync function
             PageRunAndWaitForNavigationOptions options = new PageRunAndWaitForNavigationOptions();
+
+            // URL that should be redirected to
             options.UrlString = desiredUrl;
-            var logger = _singleTestInstanceState.GetLogger();
+
             ValidatePage();
 
             try
             {
+                // Only continue if redirected to the correct page
                 await Page.RunAndWaitForNavigationAsync(async () =>
                 {
+                    // Find the password box
                     await Page.Locator(selector).WaitForAsync();
+
+                    // Fill in the password
                     await Page.FillAsync(selector, value);
+
+                    // Submit password form
                     await this.ClickAsync("input[type=\"submit\"]");
 
-                    ValidatePage();
-                    await Page.ClickAsync("[id=\"idBtn_Back\"]");
+                    // For instances where there is a 'Stay signed in?' dialogue box
+                    try
+                    {
+                        ValidatePage();
+
+                        logger.LogDebug("Checking if asked to stay signed in.");
+
+                        // Check if we received a 'Stay signed in?' box?
+                        await Page.WaitForSelectorAsync("[id=\"KmsiCheckboxField\"]");
```

Comment:
The default timeout for WaitForSelectorAsync() is 30s. I feel it's a little too long in our case.